File: refactor_patients.js
const fs = require('fs');
const path = require('path');

const targetDir = 'c:\\Users\\dayoo\\.gemini\\antigravity\\scratch\\patriotic-virtual-prod\\emr-portal';

function replaceInFile(filePath) {
    let content;
    try {
        content = fs.readFileSync(filePath, 'utf8');
    } catch (err) {
        console.error(`Could not read ${filePath}: ${err}`);
        return;
    }

    // RegEx replacements
    // (?<!...) is not supported in all JS environments but Node 24 supports it.
    // However, for compatibility I'll do a simple match and conditional replace or use lookbehind if supported.

    // Node 24 supports lookbehind.

    const replacements = [
        [/(?<!use\s+)clients/g, 'patients'],
        [/(?<!use\s+)Clients/g, 'Patients'],
        [/(?<!use\s+)client/g, 'patient'],
        [/(?<!use\s+)Client/g, 'Patient'],
        [/CLIENTS/g, 'PATIENTS'],
        [/CLIENT(?!\s+Component)/g, 'PATIENT']
    ];

    let newContent = content;
    for (const [pattern, replacement] of replacements) {
        newContent = newContent.replace(pattern, replacement);
    }

    if (newContent !== content) {
        fs.writeFileSync(filePath, newContent, 'utf8');
        console.log(`Updated: ${filePath}`);
    }
}

function walk(dir) {
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        const fullPath = path.join(dir, file);
        if (fullPath.includes('node_modules') || fullPath.includes('.next') || fullPath.includes('.git')) {
            return;
        }
        const stat = fs.statSync(fullPath);
        if (stat && stat.isDirectory()) {
            walk(fullPath);
        } else {
            if (['.ts', '.tsx', '.js', '.jsx', '.json', '.md'].includes(path.extname(fullPath))) {
                replaceInFile(fullPath);
            }
        }
    });
}

walk(targetDir);
console.log('Refactor complete.');

File: refactor_patients.py
import os
import re

def replace_in_file(file_path):
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
    except Exception as e:
        print(f"Could not read {file_path}: {e}")
        return

    # Use negative lookbehind to avoid 'use client' or 'use Client'
    # Lookbehind in Python: (?<!...)
    
    # We replace:
    # CLIENTS -> PATIENTS
    # CLIENT  -> PATIENT (but not CLIENT Component)
    # clients -> patients
    # Clients -> Patients
    # client  -> patient
    # Client  -> Patient
    
    replacements = [
        (r'(?<!use\s)clients', 'patients'),
        (r'(?<!use\s)Clients', 'Patients'),
        (r'(?<!use\s)client', 'patient'),
        (r'(?<!use\s)Client', 'Patient'),
        (r'CLIENTS', 'PATIENTS'),
        (r'CLIENT(?!\s+Component)', 'PATIENT'),
    ]
    
    new_content = content
    for pattern, replacement in replacements:
        # We need to handle case-sensitive replacements correctly
        # The lookbehind works for case-sensitive too
        new_content = re.sub(pattern, replacement, new_content)

    if new_content != content:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated: {file_path}")

def walk_and_replace(root_dir):
    for root, dirs, files in os.walk(root_dir):
        if any(x in root for x in ['node_modules', '.next', '.git']):
            continue
            
        for file in files:
            if file.endswith(('.ts', '.tsx', '.js', '.jsx', '.json', '.md')):
                replace_in_file(os.path.join(root, file))

if __name__ == "__main__":
    target = r'c:\Users\dayoo\.gemini\antigravity\scratch\patriotic-virtual-prod\emr-portal'
    walk_and_replace(target)
